### PR TITLE
Remove java backend

### PR DESCRIPTION
**Pull request by ehildenb**
This removes the java backend entirely from the repository

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -39,7 +39,6 @@ pipeline {
           options { timeout(time: 120, unit: 'MINUTES') }
           parallel {
             stage('Conformance (LLVM)') { steps {                                         sh 'make test-conformance -j4 TEST_CONCRETE_BACKEND=llvm'      } }
-            stage('Proofs (Java)')      { steps { lock("kevm-java-${env.NODE_NAME}")    { sh 'make test-prove       -j2 TEST_SYMBOLIC_BACKEND=java'    } } }
```
Comment (anvacaru) [inline Jenkinsfile:42]: Do you know if the Jenkinsfile is currently being used? Could we remove it altogether?

File: README.md [hunk 2]
```diff
@@ -44,7 +44,7 @@ Installing/Building
 
 ### K Backends
 
-There are three backends of K available: LLVM (default) for concrete execution and Haskell (default) and Java for symbolic execution.
+There are two backends of K available: LLVM for concrete execution and Haskell for symbolic execution.
```
Comment (anvacaru) [inline README.md:47]: Perhaps we should remove OpenJDK from the dependencies list as well; I can do that in #1558 .

**Review by anvacaru — APPROVED** [review]
There are two more modules, `INT-SIMPLIFICATION-JAVA` in `include/kframework/lemmas/int-simplification.k` and `LEMMAS-JAVA` in `include/kframework/lemmas/lemmas.k` . Is there a reason to exclude them?

File: README.md (hunk 2 above)
Comment (ehildenb) [reply README.md:47]: OpenJDK is needed needed for K itself.

File: Jenkinsfile (hunk 1 above)
Comment (ehildenb) [reply Jenkinsfile:42]: It's used for the master release job.

**Pull request merged**